src/h3lib/lib/polyfill.c

```diff
@@ -433,7 +433,8 @@ void iterStepPolygonCompact(IterCellsPolygonCompact *iter) {
 
         // Target res: Do a fine-grained check
         if (cellRes == iter->_res) {
-            if (mode == CONTAINMENT_CENTER || mode == CONTAINMENT_OVERLAPPING) {
+            if (mode == CONTAINMENT_CENTER || mode == CONTAINMENT_OVERLAPPING ||
+                mode == CONTAINMENT_OVERLAPPING_BBOX) {
```

Comment:
Hm. These additional checks might get the fuzzer to pass, but they miss the point of `CONTAINMENT_OVERLAPPING_BBOX`, which is to do a much faster check than `CONTAINMENT_OVERLAPPING`. The perf impact here is likely significant, since we're going from a fast bbox check to a slow set of polygon-based checks.

Are all of these additional checks needed to make the fuzzer pass? Or can we narrow down to find the check that's actually missed by `CONTAINMENT_OVERLAPPING_BBOX`?

Reply:
All but the change around line 511 seem to be necessary to prevent crashes.

Reply:
I admit that I'm not seeing any significant differences in the benchmarks after this change, assuming I'm running them correctly. I think I'm ok with this to get the new algo out the door, though I'd really like to look through the logic here and understand what's missing in the bbox check - it doesn't make a lot of sense to me conceptually that these would work when the bbox doesn't, so we must be messing up the bbox check in some way.